great_expectations/datasource/data_connector/util.py

```diff
@@ -290,6 +288,49 @@ def get_filesystem_one_level_directory_glob_path_list(
     return path_list
 
 
+def list_azure_keys(
+    azure: BlobServiceClient,
+    query_options: dict,
+    recursive: bool = False,
```

Comment:
Is the `recursive` argument used for the Azure storage use case?

Reply:
Would you mind clarifying your question? 

`recursive` is not an explicit argument in the Azure SDK but something we implement here to have greater control over the traversal of blobs in a given container. Without it, we are unable to retrieve blobs at a certain depth in the hierarchy (it by default traverses down as much as possible). 

This should be aligned with how S3 traverses its own structure through our `list_s3_keys` method (not something the user directly invokes but rather an internal method to fulfill the behavior requested of the given class).